pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -224,22 +216,22 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 			//    VolumeCapability/fsType (if present in request) must match used fsType.
 			// 3) Readonly MUST match
 			// If there is mismatch of any of above, we return ALREADY_EXISTS error.
-			existingFsType, err := determineFilesystemType(device.Path)
+			mpList, err := ns.mounter.List()
 			if err != nil {
-				return nil, err
+				return nil, status.Errorf(codes.Internal, "Failed to fetch existing mount details for %q: %v", targetPath, err)
```

Comment:
The error is not specific to the target path, is it? If yes, then the error message will be misleading - better say "Failed to fetch existing mount details while checking %q".

Reply:
> The error is not specific to the target path, is it?

The check is for the target path.

 > If so, then the error message will be misleading - better say "Failed to fetch existing mount details while checking %q".

Rephrased the error text.

Reply:
> The check is for the target path.

Yes, but the error is not *about* the error path. Usually, errors follow the format "object: operation" and deviating from that would imply some connection that isn't there. Anyway, thanks for changing it.